docs/src/examples/type_parameter.md

```diff
@@ -0,0 +1,60 @@
+# The type parameter `T` in `Quaternion{T}`
+
+The type parameter `T <: Real` in `Quaternion{T}` represents the type of real and imaginary parts of a quaternion.
+
+## Lipschitz quaternions
+By using this type parameter, some special quaternions such as [**Lipschitz quaternions**](https://en.wikipedia.org/wiki/Hurwitz_quaternion) ``L`` can be represented.
+
+```math
+L = \left\{a+bi+cj+dk \in \mathbb{H} \mid a,b,c,d \in \mathbb{Z}\right\}
+```
+
+```@setup LipschitzHurwitz
+using Quaternions
+```
+
+```@repl LipschitzHurwitz
+q1 = Quaternion{Int}(1,2,3,4)
+q2 = Quaternion{Int}(5,6,7,8)
+islipschitz(q::Quaternion) = isinteger(q.s) & isinteger(q.v1) & isinteger(q.v2) & isinteger(q.v3)
+islipschitz(q1)
+islipschitz(q2)
+islipschitz(q1 + q2)
+islipschitz(q1 * q2)
+islipschitz(q1 / q2)  # Division is not defined on L.
+q1 * q2 == q2 * q1  # non-commutative
+```
+
+## Hurwitz quaternions
+If all coefficients of a quaternion are integers or half-integers, the quaternion is called a [**Hurwitz quaternion**](https://en.wikipedia.org/wiki/Hurwitz_quaternion).
+The set of Hurwitz quaternions is defined by
+
+```math
+H = \left\{a+bi+cj+dk \in \mathbb{H} \mid a,b,c,d \in \mathbb{Z} \ \text{or} \ a,b,c,d \in \mathbb{Z} + \tfrac{1}{2}\right\}.
+```
+
+Hurwitz quaternions can be implemented with [HalfIntegers.jl](https://github.com/sostock/HalfIntegers.jl) package.
+
+```@repl LipschitzHurwitz
+using HalfIntegers
+q1 = Quaternion{HalfInt}(1, 2, 3, 4)
+q2 = Quaternion{HalfInt}(5.5, 6.5, 7.5, 8.5)
+q3 = Quaternion{HalfInt}(1, 2, 3, 4.5)  # not Hurwitz quaternion
+ishurwitz(q::Quaternion) = (isinteger(q.s) & isinteger(q.v1) & isinteger(q.v2) & isinteger(q.v3)) | (ishalfinteger(q.s) & ishalfinteger(q.v1) & ishalfinteger(q.v2) & ishalfinteger(q.v3))
```

Comment:
```suggestion
ishurwitz(q::Quaternion) = (isinteger(q.s) & isinteger(q.v1) & isinteger(q.v2) & isinteger(q.v3)) | (ishalfoddinteger(q.s) & ishalfoddinteger(q.v1) & ishalfoddinteger(q.v2) & ishalfoddinteger(q.v3))
```
`ishalfinteger` was incorrect. I'll update like this after https://github.com/sostock/HalfIntegers.jl/pull/59 is merged.

Reply:
```suggestion
ishurwitz(q::Quaternion) = (isinteger(q.s) & isinteger(q.v1) & isinteger(q.v2) & isinteger(q.v3)) | (ishalfodd(q.s) & ishalfodd(q.v1) & ishalfodd(q.v2) & ishalfodd(q.v3))
```
https://github.com/sostock/HalfIntegers.jl/pull/59#discussion_r1415705222